lib/lib-rds/src/Signer.ts

```diff
@@ -0,0 +1,61 @@
+import { Sha256 } from "@aws-crypto/sha256-js";
+import { defaultProvider } from "@aws-sdk/credential-provider-node";
+import { HttpRequest } from "@aws-sdk/protocol-http";
+import { SignatureV4 } from "@aws-sdk/signature-v4";
+import { CredentialProvider, Credentials } from "@aws-sdk/types";
+import { formatUrl } from "@aws-sdk/util-format-url";
+
+export interface SignerInit {
+  credentials?: Credentials | CredentialProvider;
+  hostname: string;
+  port: number;
+  region: string;
+  username: string;
+}
+
+export class Signer {
+  private readonly credentials: Credentials | CredentialProvider;
+  private readonly hostname: string;
+  private readonly port: number;
+  private readonly protocol: string = "https:";
+  private readonly region: string;
+  private readonly service: string = "rds-db";
+  private readonly username: string;
+
+  constructor({ credentials, hostname, port, region, username }: SignerInit) {
+    this.credentials = credentials ?? defaultProvider();
+    this.hostname = hostname;
+    this.port = port;
+    this.region = region;
+    this.username = username;
+  }
+
+  public async getAuthToken(): Promise<string> {
+    const signer = new SignatureV4({
+      service: this.service,
+      region: this.region,
+      credentials: this.credentials,
+      sha256: Sha256,
+    });
+
+    const request = new HttpRequest({
+      method: "GET",
+      protocol: this.protocol,
+      hostname: this.hostname,
+      port: this.port,
+      query: {
+        Action: "connect",
+        DBUser: this.username,
+      },
+      headers: {
+        host: `${this.hostname}:${this.port}`,
+      },
+    });
+
+    const presigned = await signer.presign(request, {
+      expiresIn: 900,
+    });
+
+    return formatUrl(presigned).replace(`${this.protocol}//`, "");
```

Comment:
This seems to be confusing. Why is the protocol removed from the full URL?

Reply:
This is a quirk of how RDS implements IAM authentication. [See the botocore notes for their description of this issue here](https://github.com/aws/lumberyard/blob/414466d63520443fafd8c27c4735fee099914384/dev/Tools/AWSPythonSDK/1.5.8/botocore/signers.py#L435).

I will add additional comments to clarify in this repo as well.

Reply:
Additional example from the CLI (note the Output without the leading protocol): https://docs.aws.amazon.com/cli/latest/reference/rds/generate-db-auth-token.html